File: reforged_16/src/main/java/top/figsq/poketip/poketip/impl/SpeciesWrapper.java
package top.figsq.poketip.poketip.impl;

import com.pixelmonmod.pixelmon.api.pokemon.species.Species;
import lombok.Getter;
import top.figsq.poketip.poketip.api.pokemon.ISpeciesWrapper;

/**
 * 实现类
 */
@Getter
public class SpeciesWrapper extends ISpeciesWrapper<Species> {
    private final Species original;

    public SpeciesWrapper(Species original) {
        if (original == null) throw new IllegalArgumentException("EnumSpecies can't be null");
        this.original = original;
    }

    @Override
    public String getName() {
        return original.getLocalizedName();
    }

    @Override
    public boolean isLegend() {
        return original.isLegendary();
    }

    @Override
    public boolean isMythical() {
        return original.isMythical();
    }

    @Override
    public int getGeneration() {
        return original.getGeneration();
    }

    @Override
    public Class<Species> getType() {
        return Species.class;
    }
}

File: cobblemon_21/src/main/kotlin/top/figsq/poketip/poketip/impl/SpeciesWrapper.kt
package top.figsq.poketip.poketip.impl

import com.cobblemon.mod.common.api.pokemon.labels.CobblemonPokemonLabels
import com.cobblemon.mod.common.pokemon.Species
import top.figsq.poketip.poketip.api.pokemon.ISpeciesWrapper

class SpeciesWrapper(
    private val species: Species
) : ISpeciesWrapper<Species>() {
    override fun getName(): String {
        return this.species.translatedName.string
    }

    override fun isLegend(): Boolean {
        return this.species.labels.contains(CobblemonPokemonLabels.LEGENDARY)
    }

    override fun isMythical(): Boolean {
        return this.species.labels.contains(CobblemonPokemonLabels.MYTHICAL)
    }

    override fun getGeneration(): Int {
        return this.species.labels.firstOrNull {
            it.startsWith("GENERATION_")
        }
            ?.substring(11)
            ?.replace("[^0-9]", "")
            ?.toInt()
            ?: -1
    }

    override fun getOriginal(): Species {
        return this.species
    }

    override fun getType(): Class<Species> {
        return Species::class.java
    }

}
